peregrine/tracking.py

```diff
@@ -357,21 +387,34 @@ def do_channel(chan, n=None, q_progress=None):
     if q_progress:
       q_progress.put(1.0 - progress)
 
-    return track_result
+    return track_result, l2c_handover_chan
 
+  # Run L1CA
+  if multi:
+    track_handover_results=pp.parmap(do_channel, channels,
+                                     show_progress=show_progress, func_progress=show_progress)
+  else:
+    track_handover_results=map(lambda (n, chan): do_channel(chan, n=n), enumerate(channels))
+  # Extract track and handover results
+  l1ca_track_results    = map(lambda x : x[0], track_handover_results)
+  l2c_handover_channels = map(lambda x : x[1], track_handover_results)
+ 
+  # Run L2C
+  logger.info("Start L2C tracking")
   if multi:
-    track_results = pp.parmap(do_channel, channels,
-                              show_progress=show_progress, func_progress=show_progress)
+    track_handover_results=pp.parmap(do_channel, l2c_handover_channels,
+                                     show_progress=show_progress, func_progress=show_progress)
   else:
-    track_results = map(
-        lambda (n, chan): do_channel(chan, n=n), enumerate(channels))
+    track_handover_results=map(lambda (n, chan): do_channel(chan, n=n), enumerate(l2c_handover_channels))
+  # Extract track results, handover results are unused 
+  l2c_track_results = map(lambda x : x[0], track_handover_results)
 
   if pbar:
     pbar.finish()
 
   logger.info("Tracking finished")
 
-  return track_results
+  return l1ca_track_results, l2c_track_results
```

Comment:
How about not return 'l2c_track_results' like this, but instead extend the 'track_results' list with L2C track results? To make it feasible your would need to extend TrackResults class with an extra field, which would tell us if the result is for L1C/A or L2C.